pkg/endpoint/policy.go

```diff
@@ -498,6 +499,230 @@ func (e *Endpoint) IngressOrEgressIsEnforced() bool {
 		e.Opts.IsEnabled(OptionEgressPolicy)
 }
 
+func (e *Endpoint) GetPortNumber(port string) (uint16, error) {
```

Comment:
What is endpoint specific about this function? Port name mapping? Needs a more specific name for sure such as `ResolvePortName()` which should be an interface defined in the policy package which the endpoint can implement.

Reply:
> Port name mapping?

Yes

>...which should be an interface defined in the policy package which the endpoint can implement.

I'm not following, why an interface?

Reply:
See comment thread of `regeneratePolicyNew()`, same reason.